app/src/main/kotlin/at/bitfire/davdroid/ui/DebugInfoActivity.kt

```diff
@@ -120,101 +167,220 @@ class DebugInfoActivity : AppCompatActivity() {
-                    .setStream(
-                        FileProvider.getUriForFile(
-                            this,
-                            getString(R.string.authority_debug_provider),
-                            zipFile
+            modelCause?.let { cause ->
+                item {
+                    CardWithImage(
+                        image = painterResource(R.drawable.undraw_server_down),
+                        title = when (cause) {
+                            is HttpException -> stringResource(if (cause.code / 100 == 5) R.string.debug_info_server_error else R.string.debug_info_http_error)
+                            is DavException -> stringResource(R.string.debug_info_webdav_error)
+                            is IOException, is IOError -> stringResource(R.string.debug_info_io_error)
+                            else -> cause::class.java.simpleName
+                        },
+                        subtitle = cause.localizedMessage,
+                        message = stringResource(
+                            if (cause is HttpException)
+                                when {
+                                    cause.code == 403 -> R.string.debug_info_http_403_description
+                                    cause.code == 404 -> R.string.debug_info_http_404_description
+                                    cause.code / 100 == 5 -> R.string.debug_info_http_5xx_description
+                                    else -> R.string.debug_info_unexpected_error
+                                }
+                            else
+                                R.string.debug_info_unexpected_error
+                        ),
+                        icon = Icons.Rounded.Info,
+                        modifier = Modifier.padding(horizontal = 8.dp, vertical = 8.dp)
+                    ) {
+                        TextButton(
+                            enabled = debugInfo != null,
+                            onClick = { shareFile(debugInfo!!) }
+                        ) {
+                            Text(
+                                stringResource(R.string.debug_info_view_details).uppercase()
+                            )
+                        }
+                    }
+                }
+            }
+            debugInfo?.let { info ->
+                item {
+                    CardWithImage(
+                        image = painterResource(R.drawable.undraw_server_down),
+                        title = stringResource(R.string.debug_info_title),
+                        subtitle = stringResource(R.string.debug_info_subtitle),
+                        icon = Icons.Rounded.BugReport,
+                        modifier = Modifier.padding(horizontal = 8.dp, vertical = 8.dp)
+                    ) {
+                        TextButton(
+                            onClick = { shareFile(info) }
```

Comment:
See above.

_Debug info / View details_ and _Logs / view logs_ should use ACTION_VIEW; the ZIP archive should be shared.